tools/flaky-test-reporter/github_issue_test.go

```diff
@@ -0,0 +1,218 @@
+		Config:             config,
+		TestStats:          tss,
+		LastBuildStartTime: &startTime,
+	}
+}
+
+func TestCreateIssue(t *testing.T) {
+	datas := []struct {
+		passed, flaky, failed, notenoughdata int
+		wantIssues                           int
+	}{
+		{197, 2, 0, 0, 1}, // flaky rate > 1%, create only 1 issue
+		{200, 2, 0, 0, 2}, // flaky rate < 1%, create issue for each
+	}
+
+	for _, d := range datas {
+		fgc := getFakeGithubIssueClient()
+		repoData := createRepoData(d.passed, d.flaky, d.failed, d.notenoughdata, int64(0))
+		fgc.processGithubIssueForRepo(repoData, make(map[string][]*flakyIssue), fakeRepo, dryrun)
+		issues, _ := fgc.client.ListIssuesByRepo(fakeOrg, fakeRepo, []string{})
+		if len(issues) != d.wantIssues {
+			t.Fatalf("2%% tests failed, got %d issues, want %d issue", len(issues), d.wantIssues)
+		}
+	}
+}
+
+func TestExistingIssue(t *testing.T) {
+	fgc := getFakeGithubIssueClient()
+	repoData := createRepoData(200, 2, 0, 0, int64(0))
+	flakyIssuesMap, _ := fgc.getFlakyIssues()
+	fgc.processGithubIssueForRepo(repoData, flakyIssuesMap, fakeRepo, dryrun)
+	existIssues, _ := fgc.client.ListIssuesByRepo(fakeOrg, fakeRepo, []string{})
+	flakyIssuesMap, _ = fgc.getFlakyIssues()
+
+	*repoData.LastBuildStartTime++
+	fgc.processGithubIssueForRepo(repoData, flakyIssuesMap, fakeRepo, dryrun)
+	issues, _ := fgc.client.ListIssuesByRepo(fakeOrg, fakeRepo, []string{})
+	if len(existIssues) != len(issues) {
+		t.Fatalf("issues already exists, got %d new issues, want 0 new issues", len(issues)-len(existIssues))
+	}
+}
+
+func TestUpdateIssue(t *testing.T) {
+	dataForTest := []struct {
+		issueState     string
+		ts             TestStat
+		passedLastTime bool
+		appendComment  bool
+		wantStatus     string
+		wantErr        error
+	}{
+		{"open", tsMapForTest["flaky"], false, true, "open", nil},
+		{"open", tsMapForTest["flaky"], true, true, "open", nil},
+		{"open", tsMapForTest["passed"], false, true, "open", nil},
+		{"open", tsMapForTest["passed"], true, true, "closed", nil},
+		{"open", tsMapForTest["failed"], false, true, "open", nil},
+		{"open", tsMapForTest["failed"], true, true, "open", nil},
+		{"open", tsMapForTest["notenoughdata"], false, true, "open", nil},
+		{"open", tsMapForTest["notenoughdata"], true, true, "open", nil},
+		{"closed", tsMapForTest["flaky"], false, true, "open", nil},
+		{"closed", tsMapForTest["flaky"], true, true, "open", nil},
+		{"closed", tsMapForTest["passed"], false, true, "closed", nil},
+		{"closed", tsMapForTest["passed"], true, true, "closed", nil},
+		{"closed", tsMapForTest["failed"], false, true, "closed", nil},
+		{"closed", tsMapForTest["failed"], true, true, "closed", nil},
+		{"closed", tsMapForTest["notenoughdata"], false, true, "closed", nil},
+		{"closed", tsMapForTest["notenoughdata"], true, true, "closed", nil},
+	}
+
+	title := "c"
+	body := "d"
+
+	for _, data := range dataForTest {
+		fgc := getFakeGithubIssueClient()
+		issue, comment := createFlakyIssue(fgc, title, body)
```

Comment:
put it in else so only call it when needed?

Reply:
done